### DISCUSSION TITLE
Where is RequireFields coming from?

### DISCUSSION THREAD

**Discussion by ziimakc** [dotansimha/graphql-code-generator#8134] [Q&A]
Maybe someone knows where is `RequireFields` are coming from? It makes `x` type `number` instead of `number | null` as it should be. And how to fix it.

```
    test(
      x: Int = 1 # it can be null if user will pass null
    ): Int!
```
```
export type QueryResolvers<
  ContextType = MercuriusContext,
  ParentType extends ResolversParentTypes['Query'] = ResolversParentTypes['Query'],
> = {
  test?: Resolver<
    ResolversTypes['smth'],
    ParentType,
    ContextType,
    RequireFields<QuerytestArgs, 'x'>
  >;
};
```

```
export type QuerytestArgs = {
  x?: InputMaybe<Scalars['Int']>;
};
```

**Comment by ardatan** [comment]
But this is the resolver signature so you will always get a value even if the user doesn't pass anything, right?

**Reply by ziimakc** [reply]
@ardatan If user will pass `null` explicitly, I will get `null` while types say it's `number`: #8135
